<?xml version="1.0"?>

<launch>

  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find igvc_gazebo)/config/igvc_joint_control.yaml" command="load"/>

  <!-- load controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner"
        args="$(find igvc_gazebo)/config/igvc_joint_control.yaml"/>

  <!-- Controller -->
  <node name="igvc_control" pkg="igvc_gazebo" type="gazebo_control" output="screen">
    <rosparam file="$(find igvc_gazebo)/config/control_coeffs.yaml" command="load"/>
    <param name="topics/motors" value="/motors"/>
    <param name="topics/encoders" value="/encoders"/>
    <param name="topics/voltage/left" value="/voltage/left"/>
    <param name="topics/voltage/right" value="/voltage/right"/>
    <param name="max_voltage" value="63.0"/>
    <param name="rate" value="60.0"/>

  </node>

</launch>
